### ISSUE TITLE
Trying to compile the UWP library but I get a fault

### ISSUE THREAD

**Issue by gordonpagan** [pauldendulk/Mapsui#156]
I am just familiarizing myself with this very promising looking library and was wanting to run it under UWP. In Visual Studio 15 with Brutile (Installed using NuGet) I get the following errors:-

> Error		The package BruTile with version 0.20.0 could not be found in C:\Users\paul\.nuget\packages\. Run a NuGet package restore to download the package.	Mapsui.Samples.Uwp			
> Error		The package Microsoft.CSharp with version 4.0.1 could not be found in C:\Users\paul\.nuget\packages\. Run a NuGet package restore to download the package.	Mapsui.UI.Uwp

			

Paul, is there a path adjustment I must make to make the download find the appropriate Module? I had a look around and I could not work out where to make the changes. I would be most grateful for your input here.

**Comment by pauldendulk** [comment]
If a create a new UWP project for Windows 10 build 10586 I can install-package Mapsui. I can build the project and in the bin folder the BruTile.dll is there. 

UWP uses the new nuget references instead of the dll references. This is why you don't see BruTile as a references. The Mapsui nuget package is itself referencing BruTile. This reference is used by the UWP build. 

Not sure why you get the error in your procedures, but perhaps you could check if this works for you?

**Comment by gordonpagan** [comment]
Dear Paul,

 

Golly that was a lightning fast reply! I will try what you suggest. I am so grateful you came back so quickly.

 

Incidentally, I have been programming since 1969 and have never really got skilled at coding but good enough for government work as they say. But when I look at what you have done, I am so impressed and envious of the scope of it. I do hope this goes somewhere for you.

 

I’ll keep you in the loop with regard to my little effort.

 

Best wishes and thank you for the warp 9 reply!

 

Gordon Pagan

**Comment by gordonpagan** [comment]
I am so sorry I just keep getting this wrong path even though BruTiles is quite clearly visible in the Object Browser. The compilation still comes back with:- 

> Error		The package BruTile with version 0.20.0 could not be found in C:\Users\paul\.nuget\packages\. Run a NuGet package restore to download the package.	Mapsui.Samples.Uwp			

Just wondered what I should try next?

**Comment by pauldendulk** [comment]
Could you tell me te steps to take from create new project? I can not reproduce it so far. The ```paul.nuget``` part is weird, so that indicates an error on my side, I just can't reproduce it. Did you start a clear new project and add Mapsui? If so, why did you install brutile, it comes with Mapsui? Or did you download the Mapsui source? In that case I also do not understand why you would install brutile.

**Comment by gordonpagan** [comment]
Just to let you know I disabled the project Mapsio.Samples.Uwp in the suite then made Mapsui.Samples.Android as my startup and it worked perfectly. Don't know why the UWP should object here but it still does when enabled.
Hope this helps.

**Comment by pauldendulk** [comment]
Are you just trying to build the Mapsui source code?

**Comment by gordonpagan** [comment]
Dear Paul,

 

I am sorry to be so irritating over this but I got the problem just doing the very few steps to prove the package compiles.

 

1)      Unzip Mapsui.zip

2)      Double click on Mapsui.sln in the Mapsui-master directory.

3)      Clicked on Build Solution to get this:-

 



 

I then unloaded Mapsui.Samples.Uwp and compiled up and got this:-

 



 

Linking it to the Nexus 7 got me this:-

 



 

I don’t if this is of any help. Here is a copy of my system for you to peruse. It maybe I am using a different version.

 

Best wishes

Gordon Pagan

PS Please let me know if you want me to try anything. You have a superb piece of work here. 

 

 

Microsoft Visual Studio Enterprise 2015

Version 14.0.25431.01 Update 3

Microsoft .NET Framework

Version 4.6.02012

 

Installed Version: Enterprise

 

Architecture and Modeling Tools   00322-80000-00000-AA310

Microsoft Architecture and Modeling Tools

    

UML® and Unified Modeling Language™ are trademarks or registered trademarks of the Object Management Group, Inc. in the United States and other countries.

 

Visual Basic 2015   00322-80000-00000-AA310

Microsoft Visual Basic 2015

 

Visual C# 2015   00322-80000-00000-AA310

Microsoft Visual C# 2015

 

Visual C++ 2015   00322-80000-00000-AA310

Microsoft Visual C++ 2015

 

AnalyzePackage   1.0

OpenCL Analyze package

 

Application Insights Tools for Visual Studio Package   7.18.00214.2

Application Insights Tools for Visual Studio

 

ASP.NET and Web Tools 2015.1 (Beta8)   14.1.11107.0

ASP.NET and Web Tools 2015.1 (Beta8)

 

ASP.NET Web Frameworks and Tools 2012.2   4.1.41102.0

For additional information, visit http://go.microsoft.com/fwlink/?LinkID=309563

 

ASP.NET Web Frameworks and Tools 2013   5.2.40314.0

For additional information, visit http://www.asp.net/

 

Code Builder - API Debugger   4.0.0.1

Enables various API debugging facilities for OpenCL applications. These include (among other features): tracing, dumping buffers & images and visualizing the active OpenCL objects during debug session and afterwards.

 

CodeBuilderSessionManager   1.0.0.0

This package is in charge of managing all code builder sessions in the VS instance. A session is made of 1 or more cl files.

 

CodeBuilderTopMenu   1.0

This package is the top menu item in VS for Code Builder. This is the parent object for all other menu items in other packages.

 

Common Azure Tools   1.8

Provides common services for use by Azure Mobile Services and Microsoft Azure Tools.

 

InstallShield 2016 Editor   

InstallShield. For more information visit the Flexera Software website at <http://www.FlexeraSoftware.com>. For customer support visit <http://support.InstallShield.com>. Copyright © 2016 Flexera Software LLC. All Rights Reserved.

 

Intel Debugger for Heterogeneous Compute   1.0

Provides debugging capabilities for compute shaders running on the Intel GPU.

 

JavaScript Language Service   2.0

JavaScript Language Service

 

JavaScript Project System   2.0

JavaScript Project System

 

Merq   1.1.14-alpha (21c13c9)

Command Bus, Event Stream and Async Manager for Visual Studio extensions.

 

Microsoft Azure Mobile Services Tools   1.4

Microsoft Azure Mobile Services Tools

 

Microsoft MI-Based Debugger   1.0

Provides support for connecting Visual Studio to MI compatible debuggers

 

Mono Debugging for Visual Studio   Mono.Debugging.VisualStudio

Support for debugging Mono processes with Visual Studio.

 

MySQL for Visual Studio   1.2.6

Data design and management tools for MySQL.  Copyright © 2007-2015 Oracle, Inc.

 

NuGet Package Manager   3.4.4

NuGet Package Manager in Visual Studio. For more information about NuGet, visit http://docs.nuget.org/.

 

OCLConverter   1.0

Convert between OpenCL and standard C/C++ projects.

 

OCLDebugConfig   1.0

Information about my package

 

OpenCL Debug Engine   1

Intel(R) OpenCL Debug Engine

 

PreEmptive Analytics Visualizer   1.2

Microsoft Visual Studio extension to visualize aggregated summaries from the PreEmptive Analytics product.

 

SQL Server Data Tools   14.0.60519.0

Microsoft SQL Server Data Tools

 

TypeScript   1.8.36.0

TypeScript tools for Visual Studio

 

Visual Studio Tools for Unity   3.0.0.1

Visual Studio Tools for Unity

 

Visual Studio Tools for Universal Windows Apps   14.0.25527.01

The Visual Studio Tools for Universal Windows apps allow you to build a single universal app experience that can reach every device running Windows 10: phone, tablet, PC, and more. It includes the Microsoft Windows 10 Software Development Kit.

 

Xamarin   4.3.0.789 (4d2ed3d)

Visual Studio extension to enable development for Xamarin.iOS and Xamarin.Android.

 

Xamarin.Android   7.1.0.43 (3a62f1e)

Visual Studio extension to enable development for Xamarin.Android.

 

Xamarin.iOS   10.4.0.128 (ba11e48)

Visual Studio extension to enable development for Xamarin.iOS.

**Comment by pauldendulk** [comment]
Downloading the zip, extracting and building just works for me. I can run all samples.

Earlier you mentioned installing brutile, this is not necessary, perhaps this caused the error. Do you have the same problem if you download from scratch? 

I also work with VS 2015 Update 3. Only the UWP update could differ, I use UWP for Windows 10 build 10586.

In your mail some information seems to be missing after 'got this...'

**Comment by gordonpagan** [comment]
Dear Paul,

 

Yes I downloaded this from scratch.

I double clicked on Mapsui.sln and did not install Brutiles. This is what I got. At this stage I have done no compilation. Brutiles do not appear in the ObjectBrowser.

 



 

Then I clicked ‘Build Solution’ and this is what I got:-

 



 

Please note that the Nuget modules for Brutile and Brutile.Desktop have been automatically loaded in. I did not add these manually. 

 

 

I agree  the versions of UWP are different and that maybe the cause. 

 

Perhaps I am just an old duffer but these days there are so many layers of abstraction it is difficult sometimes to find the cause of errors. I would like to use your hard crafted library over UWP also so I shall keep looking to see if I find where the problem might be.

 

Best wishes

Gordon Pagan

**Comment by charlenni** [comment]
Here is also no problem. UWP works.

**Comment by pauldendulk** [comment]
The UWP version I have is: 
Visual Studio Tools for Universal Windows Apps 14.0.25527.01
Could you check that?

**Comment by pauldendulk** [comment]
btw, your messages are missing information. Did you drag images into your email? That does not show here in the forum.

**Comment by gordonpagan** [comment]
Dear Paul,

 

Yes I dd. I apologize for that. I was thinking of other forums. Sorry.

 

 

Best wishes

Gordon Pagan

**Comment by gordonpagan** [comment]
Yep, that is the version I have coming up in the About box in VS15.

 

Best wishes

Gordon Pagan

**Comment by pauldendulk** [comment]
I can't think of anything else causing a build problem. Perhaps you could try this on another machine. And perhaps you could paste your android results here too.

**Comment by gordonpagan** [comment]
Paul, you have tried your best for me. It must be a problem with my VS15 implementation. As I said the android works a dream and I can demo it on my Nexus 7. I think the problem must be local to me. Thank you for your attentiveness and time you have taken to solve this. I still want to get this running on my Raspberry PI 3 so I shall persist with it and let you know what quirky feature caused this.

 

Best wishes

Gordon Pagan

**Comment by pauldendulk** [comment]
Thanks for the feedback

**Comment by pauldendulk** [comment]
@gordonpagan Perhaps you can try if your issue is fixed with the new package:
https://www.nuget.org/packages/Mapsui/0.14.0
see #160

**Comment by gordonpagan** [comment]
Hi, Paul, I am going to use your Nuget installation for development purposes but for testing I have been using the source which has samples on it. I gather the BruTile version used for your GitHub source was 0.20.0 anyway. I did not download anything from Nuget on any of this test. So, using your source from Github, what I did was unzip and double click on _Mapsui.sln_ to launch your source library then:-
1) Set Mapsui.Samples.Android as the Startup Project
2) Click Build solutions.

Here I get 2 errors:-

_**Error		The package BruTile with version 0.20.0 could not be found in C:\Users\paul\.nuget\packages\. Run a NuGet package restore to download the package.	Mapsui.Samples.Uwp**_			

**_Error		The package Microsoft.CSharp with version 4.0.1 could not be found in C:\Users\paul\.nuget\packages\. Run a NuGet package restore to download the package.	_Mapsui.UI.Uwp__**			

3) Next step is to unload both of these units just by right clicking on the uwp units _Mapsui.Samples.Uwp_ and _Mapsui.UI.Uwp_ and selecting _Unload Project_.

This goes through to the end and I able to directly debug it on the Nexus 7 I have. It works great on there and the navigation response is marvelous - better than I have ever done. It clearly works for other users too and I just don't know why it doesn't for me. I have a Raspberry Pi 3 and really want to try and run it on there also. I am sure others will too. I'll keep looking into it.

**Comment by pauldendulk** [comment]
Thanks for the detailed feedback!

**Comment by gordonpagan** [comment]
Hi Paul, I am still trying find out why I am still getting my download not to compile and link on the UWP samples download with the complete source installation as one can download from GitHub. Something I just noticed that I did not see before was that  the Mapsui.Samples.Uwp (Universal Windows) sample on my version does not have Brutile referenced in its references. All other Mapsui.Samples genres do - Android, Desktop,iOS and Portable. Is this a deliberate thing is it maybe something to do with the error which I get reported:-  _The package BruTile with version 0.20.0 could not be found in C:\Users\paul\.nuget\packages\. Run a NuGet package restore to download the package._  This is purely building the code and not linking it of course.

**Comment by pauldendulk** [comment]
This is a  new feature of vs projects. Nuget references. 2 in 1. All new projects have this feature in vs 2017. Although this should work your problem might be related.

**Comment by gordonpagan** [comment]
I just downloaded your latest source. Sure enough Brutile is there on Object Browser.  I get the same problem before using the project: Mapsui.Samples.Uwp (Universal Windows) in that the Brutile reference is missing. In all other examples, it is referenced. I take what you say about the new way of dealing with DLLs in Nuget references. Please could you tell me whether your version of Mapsui.Samples.Uwp has a reference to Brutile? I was wondering whether the reference is necessary under the new system as I don't see how this makes any sense. Thanks again Paul and sorry for this repeated question.

**Comment by pauldendulk** [comment]
It has a nuget reference. It looks like this:
![image](https://cloud.githubusercontent.com/assets/963462/24695774/59fabe1c-19e7-11e7-997c-dd5e453d1e21.png)

**Comment by gordonpagan** [comment]
Dear Paul,

 

This is a most peculiar mystery.

 

1)      I just download the package via GitHub again moments ago.

2)      I unzipped and double clicked on Mapsui.sln

3)      I just examined the references for Brutile in the Project Mapsui.Samples.Uwp (Universal Windows) and this what I see:-

 



 

And the BruTile reference is missing!

Have I taken a quantum leap into another universe here? 

As I  seem to be the sole person getting this, it must be my setup but I am bewildered what I need to adjust to get this. 

Incidentally, the Mapsui.UI.Uwp (Universal Windows) module is the same:-



 

I am at a loss what to do next.

 

If you have any advice on this I would be most grateful.

Best Wishes

Gordon Pagan
